core/src/main/java/org/springframework/security/core/context/SecurityContextHolderStrategy.java

```diff
@@ -38,6 +38,12 @@
 	 */
 	SecurityContext getContext();
 
+	/**
+	 * Peeks the current context without creating an empty context.
+	 * @return a context (may be <code>null</code>)
+	 */
+	SecurityContext peekContext();
```

Comment:
For backward compatibility, this will need to have a default implementation. Please see https://github.com/spring-projects/spring-security/issues/1890#issuecomment-860929424 for more details on the recommended implementation.